src/cockpit/channels/dbus.py

```diff
@@ -172,11 +171,54 @@ class DBusChannel(Channel):
     matches = None
     name = None
     bus = None
+    owner = None
 
     # This needs to be a fair mutex so that outgoing messages don't
     # get re-ordered.  asyncio.Lock is fair.
     watch_processing_lock = asyncio.Lock()
 
+    def add_match(self, rule, handler):
+        def sync_handler(message):
+            task = asyncio.create_task(handler(message))
+            self.tasks.add(task)
+            task.add_done_callback(self.tasks.discard)
+        self.matches.append(self.bus.add_match(rule, sync_handler))
+
+    async def setup_name_owner_tracking(self):
+        def send_owner(owner):
+            # We must be careful not to send duplicate owner
+            # notifications. cockpit.js relies on that.
+            if self.owner != owner:
+                self.owner = owner
+                self.send_message(owner=owner)
+        async def handler(message):
+            if message.get_member() == "NameOwnerChanged":
+                name, old, new = message.get_body()
+                if name == self.name:
+                    send_owner(owner=new if new != "" else None)
+        self.add_match("type='signal',sender='org.freedesktop.DBus',path='/org/freedesktop/DBus',interface='org.freedesktop.DBus'", handler)
```

Comment:
Why not add `member='NameOwnerChanged'` and an `arg0=''` for the well known name to the match rule?

Reply:
Good idea! Done.